src/overlays/actors/ovl_En_Egol/z_en_egol.c

```diff
@@ -119,99 +320,1319 @@ const ActorInit En_Egol_InitVars = {
+            gDPSetEnvColor(POLY_XLU_DISP++, 155, 255, 255, 128);
+            Matrix_Translate(this->eyePos.x, this->eyePos.y, this->eyePos.z, MTXMODE_NEW);
+            Matrix_Scale(this->laserLightScale, this->laserLightScale, this->laserLightScale, MTXMODE_APPLY);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Egol/func_80A7EBDC.s")
+            laserLightScaleMod = 10.0f;
+            laserLightAlpha = 80.0f;
+            for (i = 0; i < ARRAY_COUNT(sLightOrbColors); i++) {
+                Matrix_Push();
+                Matrix_Scale(laserLightScaleMod, laserLightScaleMod, laserLightScaleMod, MTXMODE_APPLY);
+                Matrix_ReplaceRotation(&play->billboardMtxF);
+                gDPSetPrimColor(POLY_XLU_DISP++, 0, 0, sLightOrbColors[i].r, sLightOrbColors[i].g, sLightOrbColors[i].b,
+                                laserLightAlpha);
+                gSPMatrix(POLY_XLU_DISP++, Matrix_NewMtx(play->state.gfxCtx),
+                          G_MTX_NOPUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
+                gSPDisplayList(POLY_XLU_DISP++, gLightOrbVtxDL);
+                Matrix_Pop();
+                laserLightScaleMod = 3.0f;
+                laserLightAlpha = 200.0f;
+            }
+        } else {
+            gDPSetEnvColor(POLY_XLU_DISP++, 155, 255, 255, 128);
+            Matrix_Translate(this->eyePos.x, this->eyePos.y, this->eyePos.z, MTXMODE_NEW);
+            Matrix_Scale(this->chargeLightScale, this->chargeLightScale, this->chargeLightScale, MTXMODE_APPLY);
+            Matrix_Push();
+            Matrix_Scale(1.0f, 1.0f, 1.0f, MTXMODE_APPLY);
+            Matrix_ReplaceRotation(&play->billboardMtxF);
+            Matrix_RotateZS(this->chargeLightRot, MTXMODE_APPLY);
+            gDPSetPrimColor(POLY_XLU_DISP++, 0, 0, 255, 255, 255, 255);
+            gSPMatrix(POLY_XLU_DISP++, Matrix_NewMtx(play->state.gfxCtx), G_MTX_NOPUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
+            gSPDisplayList(POLY_XLU_DISP++, gLightOrbVtxDL);
+            Matrix_Pop();
+        }
+    }
+    EnEgol_DrawEffects(this, play);
+    CLOSE_DISPS(play->state.gfxCtx);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Egol/func_80A7EC84.s")
+void EnEgol_SpawnEffect(EnEgol* this, Vec3f* pos, Vec3s* rot, s16 lifetime, f32 scale, s16 type) {
+    EnEgolEffect* effect = this->effects;
+    s16 i;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Egol/func_80A7ED14.s")
+    for (i = 0; i < ARRAY_COUNT(this->effects); i++, effect++) {
+        if (!effect->active) {
+            effect->active = true;
+            effect->scale = scale;
+            effect->pos = *pos;
+            effect->timer = lifetime;
+            effect->alpha = 255;
+            effect->type = type;
+            effect->rot.x = randPlusMinusPoint5Scaled(30000.0f);
+            effect->rot.y = randPlusMinusPoint5Scaled(30000.0f);
+            effect->rot.z = randPlusMinusPoint5Scaled(30000.0f);
```

Comment:
```suggestion
            effect->rot.x = randPlusMinusPoint5Scaled(0x7530);
            effect->rot.y = randPlusMinusPoint5Scaled(0x7530);
            effect->rot.z = randPlusMinusPoint5Scaled(0x7530);
```
Always sad to see when these end up really nice numbers in decimal

Reply:
One of these days we'll have to figure out if/how we want to handle these consistently.

Reply:
In MM at least, we're sticking with hex (for now) but can't say it's the nicest solution, just a consistent one for now